plugins/source/azure/client/client.go

```diff
@@ -50,6 +51,8 @@ type Client struct {
 	BillingProfile  *armbilling.Profile
 	BillingPeriods  map[string][]*armbilling.Period
 	BillingPeriod   *armbilling.Period
+
+	storageAccountKeys *sync.Map
```

Comment:
This is shared among clients/subscriptions and contains a single key for each storage account.